plugins/authentication/ldap/ldap.php

```diff
@@ -142,7 +141,12 @@ public function onUserAuthenticate($credentials, $options, &$response)
 				// We just accept the result here
 				try
 				{
-					$ldap->bind($ldap->escape($credentials['username'], '', LDAP_ESCAPE_DN), $credentials['password']);
+					if ($this->params->get('users_dn', '') == '') {
+						$dn = $credentials['username'];
+					} else {
+						$dn = str_replace('[username]', $ldap->escape($credentials['username'], '', LDAP_ESCAPE_DN), $this->params->get('users_dn', ''));
+					}
```

Comment:
```suggestion
					if ($this->params->get('users_dn', '') == '')
					{
						$dn = $credentials['username'];
					}
					else
					{
						$dn = str_replace('[username]', $ldap->escape($credentials['username'], '', LDAP_ESCAPE_DN), $this->params->get('users_dn', ''));
					}
```
Please mind our code style rules: https://developer.joomla.org/coding-standards/php-code.html